assembly/common/src/main/resources/bin/winsqlshell.cmd

```diff
@@ -0,0 +1,280 @@
+SET SCRIPT_DIR="%~dp0"
+SET SCRIPT_NAME=%~n0%~x0
+
+REM read user arguments
+:loop
+IF NOT [%1] == [] (
+  IF "%1" == "-U" (
+    SET URL=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-h" (
+    SET HOSTARG=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-p" (
+    SET PORTARG=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-u" (
+    SET USERARG=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-Q" (
+    SET QUOTE=1
+    SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-s" (
+    SET PASSARG=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-P" (
+    SET PROMPT=1
+    SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-S" (
+    SET SECURE=1
+    SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-k" (
+    SET PRINCIPAL=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-K" (
+    SET KEYTAB=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-w" (
+    SET WIDTH=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-f" (
+    SET SCRIPT=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-o" (
+    SET OUTPUT=%2
+    SHIFT & SHIFT
+    GOTO :loop
+  ) ELSE IF "%1" == "-q" (
+    SET QUIET=1
+    SHIFT
+    GOTO :loop
+  ) ELSE (
+    REM error, print help and exit
+    CALL :help
+    EXIT /B 1
+  )
+)
+
+REM validate the options
+IF NOT [%URL%] == [] (
+  IF NOT [%HOSTARG%] == [] (
+    ECHO Error: you cannot supply both a URL and the -h host option
+    EXIT /B 1
+  ) ELSE IF NOT [%PORTARG%] == [] (
+    ECHO Error: you cannot supply both a URL and the -p port option
+    EXIT /B 1
+  ) ELSE IF NOT [%USERARG%] == [] (
+    ECHO Error: you cannot supply both a URL and the -u user option
+    EXIT /B 1
+  ) ELSE IF NOT [%PASSARG%] == [] (
+    ECHO Error: you cannot supply both a URL and the -s password option
+    EXIT /B 1
+  ) ELSE IF %SECURE% neq 0 (
+    ECHO Error: you cannot supply both a URL and the -S ssl flag
+    EXIT /B 1
+  ) ELSE IF NOT [%PRINCIPAL%] == [] (
+    ECHO Error: you cannot supply both a URL and the -k principal option
+    EXIT /B 1
+  ) ELSE IF NOT [%KEYTAB%] == [] (
+    ECHO Error: you cannot supply both a URL and the -K keytab option
+    EXIT /B 1
+  )
+)
+
+REM TODO: check password and passarg and prompt
+
+REM check for jdk1.8 and exit if not found
+SET JAVA_EXEC=java
+java -version 1>nul 2>nul || (
+   IF NOT DEFINED JAVA_HOME (
+     ECHO no java found. %SCRIPT_NAME% requires java.
+     REM error, print help and exit
+     CALL :help
+     EXIT /B 1
+   ) ELSE (
+     SET JAVA_EXEC="%JAVA_HOME%\bin\java"
+   )
+)
+FOR /f tokens^=2-5^ delims^=.-_^" %%j IN ('java -fullversion 2^>^&1') DO @set "JVER=%%j%%k"
+IF %JVER% LSS 18 (
+   ECHO Error: java is older than 1.8. %SCRIPT_NAME% requires java 1.8
+   REM error, print help and exit
+   CALL :help
+   EXIT /B 1
+)
+
+REM set splice lib dir, for Windows we assume it is always local
+
+set SCRIPT_DIR=%SCRIPT_DIR:"=%
+SET SPLICE_LIB_DIR=%SCRIPT_DIR%lib
+echo SPLICE_LIB_DIR=%SPLICE_LIB_DIR%

```

Comment:
needed?

Reply:
First two lines are needed, last echo can be removed. I removed it.